### PR TITLE
Tests envs

### PR DESCRIPTION
**Pull request by AboudyKreidieh**
tests for environments (it was getting big, so stopping here and will continue incorporating tests on a future branch)

changes:
- replaced `AccelEnv` with `LaneChangeAccelEnv` in the highway example, as a means of making sure the branch is not broken while running examples
- bug fix to visible vehicles in `LaneChangeAccelPOEnv`
- removed the use of `Tuple` objects in `AccelEnv` to make more unittestable
- tests to observation and action space of various envs
- tests to visible vehicles on envs
- tests to components of `TestEnv`

### CODE REVIEW

File: examples/sumo/highway.py [hunk 1]
```diff
@@ -5,7 +5,8 @@
 from flow.core.params import SumoParams, EnvParams, \
     NetParams, InitialConfig, InFlows
 from flow.core.vehicles import Vehicles
-from flow.envs.loop.loop_accel import AccelEnv, ADDITIONAL_ENV_PARAMS
+from flow.envs.loop.lane_changing import LaneChangeAccelEnv, \
+    ADDITIONAL_ENV_PARAMS
```
Comment (eugenevinitsky) [inline examples/sumo/highway.py]: Why the change?

**Comment by AboudyKreidieh** [comment]
@eugenevinitsky I replaced everything with utility methods. Feel free to PR again
